### PR TITLE
[Feature][seatunnel-connector-flink-fake] Support FakeSourceStream to mock data.

### PR DESCRIPTION
**Pull request by mosence**
## Purpose of this pull request

#1879 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: docs/en/connector/source/Fake.mdx [hunk 1]
```diff
@@ -116,6 +116,39 @@ source {
     FakeSourceStream {
         result_table_name = "fake"
         field_name = "name,age"
+        //mock_data_enable = true
```
Comment (ruanwenjun) [inline docs/en/connector/source/Fake.mdx]: Please write correct doc.

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSourceStream.java [hunk 2]
```diff
@@ -73,11 +135,23 @@ public String getPluginName() {
...
+        if (!mockDataEnable) {
+            while (running) {
+                int randomNum = (int) (1 + Math.random() * NAME_ARRAY.length);
+                Row row = Row.of(NAME_ARRAY[randomNum - 1], System.currentTimeMillis());
+                ctx.collect(row);
+                Thread.sleep(TimeUnit.SECONDS.toMillis(1));
+            }
+        } else {
+            int index = 0;
+            while (running && index < mockDataSize){
```
Comment (ruanwenjun) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSourceStream.java]: This connector is used to mock a stream, I suggest we don't need to add the `mockDataBounded` config here.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]
Please revert the changes of pom.xml

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSourceStream.java (hunk 2 above)
Comment (mosence) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSourceStream.java]: I have a question, newest version(>1.12) flink api, the batch api is deprecated.
[https://nightlies.apache.org/flink/flink-docs-master/docs/dev/dataset/overview/](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/dataset/overview/).
These can both all in one, Why the stream & batch must be two connector?

File: docs/en/connector/source/Fake.mdx (hunk 1 above)
Comment (mosence) [reply docs/en/connector/source/Fake.mdx]: I‘m sorry to push the pom.xml. I will resove it.

Comment (mosence) [reply docs/en/connector/source/Fake.mdx]: The docs, let me learn some alout mdx format.

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSourceStream.java (hunk 2 above)
Comment (ruanwenjun) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSourceStream.java]: There exist some deprecated api in ST, we are developing the new ST API. Before our new API finished, it's better to keep the current API.

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSource.java [hunk 3]
```diff
@@ -62,13 +111,25 @@ public String getPluginName() {
...
+                    MockSchema.mockRowTypeInfo(mockDataSchema)
+                );
+        } else {
+            Random random = new Random();
+            return env.getBatchTableEnvironment().toDataSet(
+                env.getBatchTableEnvironment().fromValues(
+                    DataTypes.ROW(DataTypes.FIELD("name", DataTypes.STRING()),
+                        DataTypes.FIELD("age", DataTypes.INT())),
+                    Arrays.stream(NAME_ARRAY).map(n -> Row.of(n, random.nextInt(AGE_LIMIT)))
+                        .collect(Collectors.toList())), Row.class);
```
Comment (ruanwenjun) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSource.java]: Can we add a default data scheme to cover this case?

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

**Comment by ruanwenjun** [comment]
You also need to update the `LICENSE` file.

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSource.java (hunk 3 above)
Comment (mosence) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSource.java]: It good idea.  Have a moment.

File: seatunnel-dist/release-docs/NOTICE [hunk 4]
```diff
@@ -4367,6 +4367,17 @@ The Apache Software Foundation (http://www.apache.org/).
...
+
+========================================================================
+jmockdata
+  * LICENSE:
+    * license/LICENSE-jmockdata.txt (Apache License 2.0)
+  * HOMEPAGE:
+    * https://github.com/jsonzou/jmockdata
+
+========================================================================
+
```
Comment (ruanwenjun) [inline seatunnel-dist/release-docs/NOTICE]: It seems jmockdata doesn't have a NOTICE file in their repo.
https://github.com/jsonzou/jmockdata

Comment (mosence) [reply seatunnel-dist/release-docs/NOTICE]: OK, my understanding of notice is wrong.
And something exception in e2e, I resolve it later

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSource.java [hunk 5]
```diff
@@ -62,13 +71,15 @@ public String getPluginName() {
...
     @Override
     public DataSet<Row> getData(FlinkEnvironment env) {
-        Random random = new Random();
-        return env.getBatchTableEnvironment().toDataSet(
-            env.getBatchTableEnvironment().fromValues(
-                DataTypes.ROW(DataTypes.FIELD("name", DataTypes.STRING()),
-                    DataTypes.FIELD("age", DataTypes.INT())),
-                Arrays.stream(NAME_ARRAY).map(n -> Row.of(n, random.nextInt(AGE_LIMIT)))
-                    .collect(Collectors.toList())), Row.class);
+        List<Row> dataSet = new ArrayList<>(0);
```
Comment (ruanwenjun) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSource.java]:
```suggestion
        List<Row> dataSet = new ArrayList<>(mockDataSize);
```

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/pom.xml [hunk 6]
```diff
@@ -49,6 +49,11 @@
             <groupId>org.apache.flink</groupId>
             <artifactId>flink-streaming-java_${scala.binary.version}</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.github.jsonzou</groupId>
+            <artifactId>jmockdata</artifactId>
+            <version>4.3.0</version>
```
Comment (ruanwenjun) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/pom.xml]: Please use `dependencyManagement` to manage the dependency version.

File: docs/en/connector/source/Fake.mdx [hunk 7]
```diff
@@ -127,6 +221,7 @@ source {
     FakeSource {
         result_table_name = "fake"
         field_name = "name,age"
+        mock_data_size = 100 // data will mock until count is 100.
```
Comment (ruanwenjun) [inline docs/en/connector/source/Fake.mdx]:
```suggestion
        mock_data_size = 100 // will generate 100 rows mock data.
```

File: docs/en/connector/source/Fake.mdx [hunk 8]
```diff
@@ -111,11 +173,43 @@ The generated data is as follows, randomly extract the string from the `content`
...
+        //    mock_config = { //see https://github.com/jsonzou/jmockdata, next is all kind config, not all configurations are required
+        //       byte_range = [0,127]  // byte value will mock range between 0 and 127.
+        //       boolean_seed = [true, true, false] // boolean  value will mock random in this array.
+        //       char_seed = ['a','b','c'] // char value will mock random in this array.
+        //       date_range = ["1970-01-01", "2100-12-31"] // date value will mock random in this array.
+        //       decimal_scale = 2 // decimal's scale is 2
+        //       double_range = [0.0, 10000.0] // double value will mock range between 0.0 and 10000.0.
+        //       float_range = [0.0, 10000.0] // float value will mock range between 0.0 and 10000.0.
+        //       int_range = [0, 100] // int value will mock range between 0 and 100.
+        //       long_range = [0, 100000] // long value will mock range between 0 and 100000.
```
Comment (ruanwenjun) [inline docs/en/connector/source/Fake.mdx]: Please remove these comments.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**